### ISSUE TITLE
Generate code from the grammar constructs (Code Generation)

### ISSUE THREAD

**Issue by multimeric** [scottfrazer/hermes#55]
Given a hermes Grammar and values for each terminal, is it possible to perform code generation? I ask because I'm interested in writing code that generates WDL (which uses a hermes grammar), but I can't see any easy or documented way to do this.

**Comment by scottfrazer** [comment]
Hey sorry!  For some reason I wasn't even aware of issues being filed on this project!

I assume you probably don't need this anymore... but in case you do, I'm not sure I understand your use case.  The only input needed for code generation is the grammar

**Comment by multimeric** [comment]
I kind of still do. I think I was hoping to do something like: generate a new `task` with name `x`, containing a list of declarations `y`, based on the grammar. Is this possible somehow?

**Comment by scottfrazer** [comment]
If I understand correctly:

Generate `wdl_parser.py`
```
$ hermes generate wdl.hgr
```

then `example.py`:
```
from wdl_parser import parse
tree = parse('''
    task y {
        Int i
        String s
    }
''')
print(tree.ast().dumps(indent=2))
```

```
$ python example.py
(Namespace:
  imports=[],
  body=[
    (Task:
      name=<string:2:10 identifier "eQ==">,
      declarations=[
        (Declaration:
          type=<string:3:9 type "SW50">,
          postfix=None,
          name=<string:3:13 identifier "aQ==">,
          expression=None
        ),
        (Declaration:
          type=<string:4:9 type "U3RyaW5n">,
          postfix=None,
          name=<string:4:16 identifier "cw==">,
          expression=None
        )
      ],
      sections=[]
    )
  ]
)
```

**Comment by multimeric** [comment]
Right, but I want to do the opposite of this. Here, you're parsing a WDL string to hermes data structures, but I want to provide hermes data structures and generate WDL using the grammar. Is this possible?

**Comment by scottfrazer** [comment]
It's possible in theory but there's no way to do this with the current code.

If you wanted to write it yourself, you could recursively walk the AST and output source code for each  node.
